File: backend/src/main/java/br/com/ifsp/classify/dtos/create/StudentCreateDTO.java
package br.com.ifsp.classify.dtos.create;

import java.time.LocalDate;
import java.util.List;

public record StudentCreateDTO(
        String name,
        LocalDate birthDate,
        LocalDate registrationDate,
        String email,
        String telephone,
        String address,
        String neighborhood,
        String school,
        Integer grade,
        Boolean referral,
        String referrerName,
        List<GuardianCreateDTO> guardians
) {}

File: frontend/src/shared/dtos/student/StudentCreateDTO.ts
import type { TelephoneCreateDTO } from "../telephone/TelephoneCreateDTO";
import type { GuardianCreateDTO } from "../guardian/GuardianCreateDTO";

export interface StudentCreateDTO {
    name: string;
    birthDate: string;
    email: string;
    cpf: string;
    registrationDate: string;
    telephones: TelephoneCreateDTO[];
    guardians: GuardianCreateDTO[];
}
